utilities/repostatus.py

```diff
@@ -0,0 +1,64 @@
+import os
+import pandas as pd
+import json
+
+# directory of the current script
+script_dir = os.path.dirname(os.path.abspath(__file__))
+
+# path to the repository
+REPO_DIR = os.path.dirname(script_dir)
+
+CODE_CONTRIBUTIONS_FILE = os.path.join(REPO_DIR, "doc", "code_contributions_record.csv")
+ALGORITHMS_FILE = os.path.join(REPO_DIR, "tests", "IVIMmodels", "unit_tests", "algorithms.json")
+SOURCE_FOLDER = os.path.join(REPO_DIR, "src", "original")
+WRAPPED_FOLDER = os.path.join(REPO_DIR, "src", "standardized")
+
+# Read the CSV file
+df = pd.read_csv(CODE_CONTRIBUTIONS_FILE)
+
+unique_subfolders = df['subfolder'].unique().tolist()
+
+# Read the JSON file
+with open(ALGORITHMS_FILE, 'r') as f:
+    algorithms_data = json.load(f)
+
+# list of all algorithms from the JSON file
+all_algorithms = algorithms_data['algorithms']
+
+# Check if both code_contributions_file matches with source folder
+for subfolder in unique_subfolders:
+    subfolder_path = os.path.join(SOURCE_FOLDER, subfolder)
+    if not os.path.exists(subfolder_path):
+        print(f"Warning: Subfolder '{subfolder}' does not exist in the source folder.")
+
+# Add column 'Tested' to the DataFrame if it starts with that of subfolder
+df['Tested'] = df.apply(lambda row: 'Yes' if any(algorithm.startswith(row['subfolder'].split('_')[0]) for algorithm in all_algorithms) else 'No', axis=1)
```

Comment:
I'm just looking at this logic and it seems like it's just matching everything before the first `_`. That's not specific because that's usually the author's initials and they can have several algorithms.

Looking at it a little more, this could become fancier, but maybe we actually want this column in our csv file. So another approach would be to just manually go through and find the matches and add a "Wrapped" column to the csv file that contains what you'd see in the algorithms.json.

Then that would reduce what is needed in here to just having a sanity check that the wrapped folders match the csv. And it would make checking for tested much easier because it would be a simple string match.

How does that sound?

Reply:
do i need to manually update the code_contribution_csv file or write a script to read algorithms from algorithms.json and write 'wrapped' in csv file   and automatically update whenever there is change in algorithms. json file

Reply:
Add the column manually. That column is not the same as what's in algorithm.json. That column would be the wrapped name, so if an algorithm is wrapped it should have an entry there. The algorithm.json is what is being tested. In theory everything wrapped should be tested but it does need to be added and confirmed it doesn't fail. So they're not the same thing.